src/sscg_generator_cli.erl

```diff
@@ -33,83 +32,84 @@ set_log_level(#{verbose := 1}) ->
 set_log_level(#{}) ->
     ok.
 
+-doc "Aborts the program with a red formatted error message".
+-spec abort(Format :: string(), Args :: list()) -> Result :: no_return().
 abort(Format, Args) ->
     io:format("~s~n", [color:red(io_lib:format(Format, Args))]),
     erlang:halt(1).
 
+-doc #{equiv => print(Text, [])}.
 print(Text) -> print(Text, []).
+
+-doc "Prints a formatted message to the console".
+-spec print(string(), [term()]) -> ok.
 print(Format, Args) ->
     io:format(Format ++ "~n", Args).
 
+-doc "Prompts the user for input".
+-spec input(string()) -> binary().
 input(Prompt) ->
     case io:get_line(Prompt ++ " ") of
         eof ->
             "";
         {error, Reason} ->
             abort("Error reading input: ~p", [Reason]);
         Data ->
-            re:replace(Data, "^[[:space:]]*+|[[:space:]]*+$", <<>>,
-                [global, {return, binary}]
-            )
+            re:replace(Data,
+                       "^[[:space:]]*+|[[:space:]]*+$",
+                       <<>>,
+                       [global, {return, binary}])
     end.
 
+-doc "Prompts a yes/no-question and returns a boolean based on the response".
+-spec confirm(string()) -> boolean().
```

Comment:
I'd drop the regex in this function too. Just trim the response and check whether or not it's exactly "yes". The current implementation would match on `"I wish I could say yes but I will say no instead"`.

I'd also check if there are libraries or built-in OTP modules that handle these CLI conversations with the user for you.